TeleportEverything/HarmonyPatches.cs

```diff
@@ -194,9 +199,29 @@ public class TeleportTo_Patch
         [HarmonyPatch("UpdateTeleport")]
         public class UpdateTeleport_Patch
         {
-            static void  Postfix( Player __instance, float dt)
+            static void  Postfix( Player __instance, ref bool ___m_teleporting, float dt)
             {
-                UpdateDelayTimer(dt);
+                if (!ZNetScene.instance.IsAreaReady(__instance.m_teleportTargetPos))
+                    return;
+
+                if (!___m_teleporting && teleportTriggered)
+                {
+                    UpdateDelayTimer(dt);
+                }
+            }
+        }
+
+        public static DelayedAction delayedAction;
+        [HarmonyPatch(typeof(Player), nameof(Player.Awake))]
+        public class PlayerAwake_Patch
+        {
+            static void Postfix(Player __instance)
+            {
+                PlayerController controller = __instance.GetComponent<PlayerController>();
+                if (delayedAction == null)
+                {
+                    delayedAction = controller.gameObject.AddComponent<DelayedAction>();
```

Comment:
What's the purpose of using the playercontroller object here?

Reply:
actually I added the component DelayedAction to PlayerController just to instance the DelayedAction object as it inherits from MonoBehaviour

Reply:
Ohh now I see what you are talking about, I could've attached it directly into the player obj. I just refactored it to insert delayed action into the game object. Let me know what you think.

Reply:
Got it.